subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidateTaskProperties.java

```diff
@@ -105,14 +110,24 @@
  * @since 3.0
  */
 @CacheableTask
-@SuppressWarnings("WeakerAccess")
-public class ValidateTaskProperties extends ConventionTask implements VerificationTask {
-    private FileCollection classes;
-    private FileCollection classpath;
-    private RegularFileProperty outputFile = getProject().getObjects().fileProperty();
+@SuppressWarnings("deprecation")
+public class ValidateTaskProperties extends ConventionTask implements VerificationTask, org.gradle.plugin.devel.tasks.internal.ValidateTaskPropertiesBackwardsCompatibleAdapter {
+    private final ConfigurableFileCollection classes;
+    private final ConfigurableFileCollection classpath;
+    private final RegularFileProperty outputFile;
+    private final Property<Boolean> enableStricterValidation;
     private boolean ignoreFailures;
     private boolean failOnWarning;
```

Comment:
We should change these to properties as well.

Reply:
I ended up using `boolean` for all the properties. If we want to use `Property<Boolean>`, then:
- This is a breaking change
- We cannot implement `VerificationTask` any more.

Or I would need to come up with some new property names for the `Property<Boolean>` setters...